PostHog/PostHogContext.swift

```diff
@@ -53,30 +53,73 @@ class PostHogContext {
+        var isiOSAppOnMac = false
+        #if os(iOS)
+            if #available(iOS 14.0, *) {
+                isiOSAppOnMac = ProcessInfo.processInfo.isiOSAppOnMac
+            }
+        #endif
+
+        // iPad app running on Mac Catalyst
+        #if targetEnvironment(macCatalyst)
+            let isMacCatalystApp = true
+        #else
+            let isMacCatalystApp = false
+        #endif
+
+        properties["$is_ios_running_on_mac"] = isiOSAppOnMac
+        properties["$is_mac_catalyst_app"] = isMacCatalystApp
+
         #if os(iOS) || os(tvOS)
             let device = UIDevice.current
             // use https://github.com/devicekit/DeviceKit
-            properties["$device_name"] = device.model
-            properties["$os_name"] = device.systemName
-            properties["$os_version"] = device.systemVersion
-
-            var deviceType: String?
-            switch device.userInterfaceIdiom {
-            case UIUserInterfaceIdiom.phone:
-                deviceType = "Mobile"
-            case UIUserInterfaceIdiom.pad:
-                deviceType = "Tablet"
-            case UIUserInterfaceIdiom.tv:
-                deviceType = "TV"
-            case UIUserInterfaceIdiom.carPlay:
-                deviceType = "CarPlay"
-            case UIUserInterfaceIdiom.mac:
-                deviceType = "Desktop"
-            default:
-                deviceType = nil
-            }
-            if deviceType != nil {
-                properties["$device_type"] = deviceType
+            let processInfo = ProcessInfo.processInfo
+
+            if isMacCatalystApp || isiOSAppOnMac {
+                let underlyingOS = device.systemName
+                let underlyingOSVersion = device.systemVersion
+                let macOSVersion = processInfo.operatingSystemVersionString
+
+                if isMacCatalystApp {
+                    let processInfo = ProcessInfo.processInfo
+                    properties["$os_name"] = "macOS \(processInfo.operatingSystemVersionString)" // eg Version 14.2.1 (Build 23C71)
+                    let osVersion = processInfo.operatingSystemVersion
+                    properties["$os_version"] = "\(osVersion.majorVersion).\(osVersion.minorVersion).\(osVersion.patchVersion)"
+                } else {
+                    properties["$os_name"] = device.systemName
+                    properties["$os_version"] = device.systemVersion
+                }
+                // device.userInterfaceIdiom reports .pad here, so we use a static value instead
+                // - For an app deployable on iPad, the idiom type is always .pad (instead of .mac)
+                // 
+                // Source: https://developer.apple.com/documentation/apple-silicon/adapting-ios-code-to-run-in-the-macos-environment#Handle-unknown-device-types-gracefully
+                properties["$device_type"] = "Desktop"
+                properties["$device_name"] = processInfo.hostName
```

Comment:
what does hostName return here? this is about the device name (Model), and not the user's name or device's nickname

Reply:
this should be the info of the physical device.
So if we're using `Host.current().localizedName` because `device.model` isn't possible, then lets keep it